### PR TITLE
Fix: Invalid relative .md links

### PR DESCRIPTION
**Pull request by niklasR**
There are a couple of different ways how links are done in this repo:
- The [Project Reference](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/documentation/copy/en/project-config/Project%20References.md) links using an absolute http URL; which is what I've done in this PR for the broken links
- The [Advanced Types](https://github.com/microsoft/TypeScript-Website/blob/v2/packages/documentation/copy/en/reference/Advanced%20Types.md) links with a relative path to the enum.html, but using the published structure paths (i.e. flattened folders and `.html`, instead of `.md`)
- These two files ("Typescript 3.0" and "Migrating from JavaScript") are the only files linking to relative .md paths; both of which are broken on the published website.

### CODE REVIEW

File: packages/documentation/copy/en/release-notes/TypeScript 3.0.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ TypeScript 3.0 introduces a new concept of project references. Project reference
 
 TypeScript 3.0 also introduces a new mode for tsc, the `--build` flag, that works hand-in-hand with project references to enable faster TypeScript builds.
 
-See [Project References handbook page](../Project%20References.md) for more documentation.
+See [Project References handbook page](https://www.typescriptlang.org/docs/handbook/project-references.html) for more documentation.
```
Comment (orta) [inline packages/documentation/copy/en/release-notes/TypeScript 3.0.md]:
```suggestion
See [Project References handbook page](/docs/handbook/project-references.html) for more documentation.
```

File: packages/documentation/copy/en/tutorials/Migrating from JavaScript.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ It's important to note that ts-loader will need to run before any other loader t
 The same goes for [awesome-typescript-loader](https://github.com/TypeStrong/ts-loader), another TypeScript loader for Webpack.
 You can read more about the differences between the two [here](https://github.com/s-panferov/awesome-typescript-loader#differences-between-ts-loader).
 
-You can see an example of using Webpack in our [tutorial on React and Webpack](./React%20&%20Webpack.md).
+You can see an example of using Webpack in our [tutorial on React and Webpack](https://www.typescriptlang.org/docs/handbook/react-&-webpack.html).
```
Comment (orta) [inline packages/documentation/copy/en/tutorials/Migrating from JavaScript.md]:
```suggestion
You can see an example of using Webpack in our [tutorial on React and Webpack](/docs/handbook/react-&-webpack.html).
```

**Comment by orta** [comment]
Thanks!

**Comment by niklasR** [comment]
@orta I've just added a commit to align other links in the documentation folder to use the same format, just so they are all aligned. Hope that fits with your thinking.

**Comment by orta** [comment]
Even better 👍🏻

**Pull request merged**